corehq/apps/userreports/sql/data_source.py

```diff
@@ -27,7 +27,7 @@ def engine_id(self):
 
     @property
     def filters(self):
-        return filter(None, [fv.to_sql_filter() for fv in self._filter_values.values()])
+        return [_f for _f in [fv.to_sql_filter() for fv in self._filter_values.values()] if _f]
```

Comment:
Does this need the inner `[]`?

Reply:
to second this, seems like it could be  `[fv.to_sql_filter() for fv in self._filter_values.values() if fv.to_sql_filter()]`

Reply:
that could be bad if `to_sql_filter` is expensive since that would call it twice

Reply:
I'd prefer not to cal ```to_sql_filter``` twice for that reason.  Maybe ```list(filter(None, [...]))``` would actually be more clear?

Reply:
Brackets within brackets seems clearer than `filter(None, [...])`

Reply:
going to go with brackets within brackets as is